ChaosMod/Effects/db/Player/PlayerZoomZoomCam.cpp

```diff
@@ -0,0 +1,72 @@
+/*
+    Effect by Moxi, based on code from DrUnderscore & ProfessorBiddle
+*/
+
+#include <stdafx.h>
+static Camera zoomCamera = 0;
+float camZoom = 80.f;
+const float camZoomRate = 0.6f;
+int n = 0;
+
+static void UpdateCamera()
+{
+    auto coord = CAM::GET_GAMEPLAY_CAM_COORD();
+    auto rot = CAM::GET_GAMEPLAY_CAM_ROT(2);
+    CAM::SET_CAM_PARAMS(zoomCamera, coord.x, coord.y, coord.z, rot.x, rot.y, rot.z, camZoom, 0, 1, 1, 2);
+}
+
+static void OnStart()
+{
+    zoomCamera = CAM::CREATE_CAM("DEFAULT_SCRIPTED_CAMERA", 1);
+    CAM::RENDER_SCRIPT_CAMS(true, true, 10, 1, 1, 1);
+}
+
+static void OnTick()
+{
+    static DWORD64 lastTick = 0;
+    DWORD64 curTick = GET_GAME_TIMER();
+
+    if (curTick > lastTick + 5)
+    {
+        lastTick = curTick;
+
+        if (camZoom < 11)
+        {
+            n = 0;
+        }
+        if (camZoom > 119)
+        {
+            n = 1;
+        }
+        if (n == 0)
+        {
+            camZoom += camZoomRate;
+            CAM::SET_CAM_ACTIVE(zoomCamera, true);
+            UpdateCamera();
+        }
+        if (n == 1)
+        {
+            camZoom -= camZoomRate;
+            CAM::SET_CAM_ACTIVE(zoomCamera, true);
+            UpdateCamera();
```

Comment:
„SET_CAM_ACTIVE“ and „UpdateCamera“ can be outside of the ifs, as they get executed either way 
Also they need to be executed every tick not every fifth or it would result in stuttering

Reply:
Thank you for that. I did notice a bit of a stutter. I was going to have a look at it to solve it. You did it for me thank you! It's making a lot more sense to write it outside of the ifs.